File: trunk/esper/src/main/java/net/esper/eql/db/DatabaseConnectionFactory.java
package net.esper.eql.db;

import java.sql.Connection;

/**
 * Factory for new database connections.
 */
public interface DatabaseConnectionFactory
{
    /**
     * Creates a new database connection.
     * @return new connection
     * @throws DatabaseConfigException throws to indicate a problem getting a new connection
     */
    public Connection getConnection() throws DatabaseConfigException;
}

File: NEsper/eql/db/DatabaseConnectionFactory.cs
using System;
using System.Data.Common;

namespace net.esper.eql.db
{
	/// <summary>
	///  Factory for new database connections.
	/// </summary>
	
	public interface DatabaseConnectionFactory
	{
		/// <summary> Creates a new database connection.</summary>
		/// <returns> new connection
		/// </returns>
		/// <throws>  DatabaseConfigException throws to indicate a problem getting a new connection </throws>

		DbConnection Connection { get ; }
	}
}
